browser/src/canvas/sections/URLPopUpSection.ts

```diff
@@ -111,18 +116,47 @@ class URLPopUpSection extends HTMLObjectSection {
 
 		document.getElementById(this.removeButtonId).onclick = () => {
 			app.map.sendUnoCommand('.uno:RemoveHyperlink', params);
+			URLPopUpSection.closeURLPopUp();
 		};
 	}
 
+	reLocateArrow(arrowAtTop: boolean) {
+		if (arrowAtTop) this.arrowDiv.classList.add('reverse');
+		else this.arrowDiv.classList.remove('reverse');
+
+		/*
+			Normally, the documentPosition sent from the core side nicely positions the arrow.
+			We will check if we reposition the popup.
+			If the arrow position falls outside of the popup, we will put it on the edge.
+		*/
+		const clientRect = this.getPopUpBoundingRectangle();
+		const arrowHalfWidth = 10;
+		let arrowCSSLeft = this.sectionProperties.documentPosition.pX - this.documentTopLeft[0] + this.containerObject.getDocumentAnchor()[0] - arrowHalfWidth;
+		arrowCSSLeft /= app.dpiScale;
+		arrowCSSLeft += document.getElementById('canvas-container').getBoundingClientRect().left; // Add this in case there is something on its left.
+		arrowCSSLeft -= clientRect.left;
+		this.arrowDiv.style.left = (arrowCSSLeft > 6 ? arrowCSSLeft : 6) + 'px'; // 6: padding of the popup.
```

Comment:
Instead of a comment, can we put 6 as const variable? it gives a label and share value in all places.
const popupPadding = 6;